### ISSUE TITLE
NickServ: detect when network supports SASL and suggest loading the SASL module instead

### ISSUE THREAD

**Issue by Mikaela** [znc/znc#1230]
It would make the module more useful if it could discover what networks supoort sasl during capability negotiation or performing cap ls (maybe even 3.2 to see what the mechanisms are) and suggest loading the sals lodule instead saying that it identifies you before being visible to the network (so a lot faster than nickserv does). 

I think it should be automatic without switch to disable the behaviour except that it could be quiet if sasl was also loaded.

**Comment by GLolol** [comment]
> I think it should be automatic without switch to disable the behaviour except that it could be quiet if sasl was also loaded.

If I'm interpreting this correctly, automatic `*sasl` setup when `*nickserv` password is set would be nice. But how would ZNC guess the SASL username?

Personally I prefer having both modules loaded in a setup so initial connections use SASL, but `*nickserv` will still identify me in cases where services go down and lose my login status.

**Comment by Mikaela** [comment]
> If I'm interpreting this correctly, automatic *sasl setup when *nickserv password is set would be nice. But how would ZNC guess the SASL username?

No, only telling *This network supports SASL so you might want to load the SASL module in order to identify before you are visible on the network (as opposed to this module which often identifies you after you are already joined to channels).*

> Personally I prefer having both modules loaded in a setup so initial connections use SASL, but *nickserv will still identify me in cases where services go down and lose my login status.

This is what I mean with *I think it should be automatic without switch to disable the behaviour except that it could be quiet if sasl was also loaded.* or in other words, always say that unless SASL is also loaded.

**Comment by GLolol** [comment]
> or in other words, always say that unless SASL is also loaded.

Couldn't it just be announced the first time the module's loaded?

**Comment by Mikaela** [comment]
> Couldn't it just be announced the first time the module (*nickserv) is loaded?

No as the user might not be connected to the network at that time and network that doesn't have SASL today could get SASL tomorrow.

**Comment by GLolol** [comment]
Or the first time the SASL cap is received but `*sasl` isn't loaded... I disagree with having the notice always on, it seems too naggy to me.

**Comment by Mikaela** [comment]
Naggy is the point and there is no reason to nag when sasl is loaded so it can well do it until thatm

**Comment by kylef** [comment]
> Personally I prefer having both modules loaded in a setup so initial connections use SASL, but *nickserv will still identify me in cases where services go down and lose my login status.

The SASL module could use [`cap-notify`](http://ircv3.net/specs/extensions/cap-notify-3.2.html) so that it would know when SASL is re-introduced and can re-authenticate over SASL.

**Comment by Mikaela** [comment]
> The SASL module could use cap-notify so that it would know when SASL is re-introduced and can re-authenticate over SASL.

https://github.com/znc/znc/issues/949
